relayer/chains/evm/listener.go

```diff
@@ -55,28 +57,30 @@ func (p *Provider) Listener(ctx context.Context, lastSavedHeight uint64, blockIn
 		subscribeStart = time.NewTicker(time.Second * 1)
 		latestHeight   = p.latestHeight(ctx)
 		concurrency    = p.GetConcurrency(ctx, startHeight, latestHeight)
-		resetChannel   = make(chan struct{})
+		errChan        = make(chan error)
 	)
 
 	for {
 		select {
 		case <-ctx.Done():
 			p.log.Debug("evm listener: done")
 			return nil
-		case <-resetChannel:
+		case err := <-errChan:
+			if p.isConnectionError(err) {
```

Comment:
The new change seems to check for certain errors and only reconnecting the client for that case wheres earlier for any packet in channel, it reconnected. Can you explain a bit on the main issue and how this filter fixes the issue ?

Reply:
Previously we reconnected in any case which is unnecessary most of the time.

Analysing the logs it appears we only need to reconnect in case of tcp related errors.

Reply:
Can you update the description in that case so it becomes more clear what we are handling on this ticket.

Reply:
Alright.